src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FallThroughCheck.java

```diff
@@ -486,46 +518,53 @@ private boolean checkSynchronized(final DetailAST synchronizedAst, boolean useBr
      * </pre>
      *
      * @param currentCase AST of the case that falls through to the next case.
-     * @param nextCase AST of the next case.
      * @return True if a relief comment was found
      */
-    private boolean hasFallThroughComment(DetailAST currentCase, DetailAST nextCase) {
-        boolean allThroughComment = false;
-        final int endLineNo = nextCase.getLineNo();
-
-        if (matchesComment(reliefPattern, endLineNo)) {
-            allThroughComment = true;
+    private boolean hasFallThroughComment(DetailAST currentCase) {
+        final DetailAST nextCase = currentCase.getNextSibling();
+        final DetailAST startAST;
+        final DetailAST stopAST;
+
+        if (nextCase.getType() == TokenTypes.CASE_GROUP) {
+            startAST = currentCase.getLastChild();
+            stopAST = nextCase.getLastChild();
         }
         else {
-            final int startLineNo = currentCase.getLineNo();
-            for (int i = endLineNo - 2; i > startLineNo - 1; i--) {
-                final int[] line = getLineCodePoints(i);
-                if (!CodePointUtil.isBlank(line)) {
-                    allThroughComment = matchesComment(reliefPattern, i + 1);
-                    break;
-                }
-            }
+            // When we have last case group of switch statement then comment might
+            // be the sibling of that case, or it might be as child of that last
+            // case group AST, so we need all the comment from last case group
+            // till the end of switch case to verify if there is relief comment.
+            startAST = currentCase.getNextSibling();
+            stopAST = currentCase.getParent().getLastChild();
```

Comment:
So this means we do a "scan" for the comments? So user can put a comment in an unnatural place (goes against documentation) which will suppress violations unintentionally?

Reply:
I have removed it.